src/compiler/checker/lines_0_1000.rs

```diff
@@ -160,40 +160,36 @@ pub fn create_type_checker<TTypeCheckerHost: TypeCheckerHost>(
             )
             .into(),
     ));
-    type_checker.any_type = Some(Rc::new(
+    type_checker.any_type = Some(
```

Comment:
Like with AST node creation, always use `.into()` (which encapsulates initializing the `_type_wrapper` reference on `BaseType`) when wrapping in an `Rc<Type>`